House-of-glass/src/Contexts/Cart/CartContext.jsx

```diff
@@ -0,0 +1,169 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+import axios from "axios";
+import { cartReducer } from "./CartReducer";
+import Toast from "components/Toast/Toast";
+
+const CartContext = createContext();
+
+const useCart = () => useContext(CartContext);
+
+const CartProvider = ({ children }) => {
+  const initialCartState = { cart: [], cartCount: 0, cartSum: 0 };
+  const [cartState, cartDispatch] = useReducer(cartReducer, initialCartState);
+
+  const token = localStorage.getItem("Token");
+  const header = {
+    headers: {
+      authorization: token,
+    },
+  };
+
+  const getCart = async () => {
+    try {
+      const response = await axios.get("/api/user/cart", header);
+      if (response.status === 200) {
+        cartDispatch({ type: "GET_CART", payload: response?.data?.cart });
```

Comment:
```js
payload: response?.data?.cart
```
What does this mean @srejitk? Is this a check for the data that is coming from the response or replacing it with a substitute

Reply:
Hey Roshan, so this is called as Optional Chaining, the purpose of which being that when accessing nested values instead of throwing an error on nullish values it'll return undefined when there's no data.